tests/cli_tests/zboxcli_create_allocation_free_storage_test.go

```diff
@@ -0,0 +1,474 @@
+		require.Nil(t, err, "Error occurred when retrieving new owner wallet")
+
+		// register this wallet
+		output, err = registerWallet(t, configPath)
+		require.Nil(t, err, "registering wallet failed", err, strings.Join(output, "\n"))
+
+		marker := climodel.FreeStorageMarker{
+			Recipient:  recipientWallet.ClientID,
+			FreeTokens: 5,
+			Timestamp:  time.Now().Unix(),
+		}
+
+		forSignatureBytes, err := json.Marshal(&marker)
+		require.Nil(t, err, "Could not marshal marker")
+
+		data := hex.EncodeToString(forSignatureBytes)
+		marker.Signature = sign(t, data, assignerWallet)
+		marker.Assigner = assignerWallet.ClientID
+
+		forFileBytes, err := json.Marshal(marker)
+		require.Nil(t, err, "Could not marshal marker")
+
+		markerFile := "./config/" + recipientCorrect + "_MARKER.json"
+
+		err = os.WriteFile(markerFile, forFileBytes, 0600)
+		require.Nil(t, err, "Could not write file marker")
+
+		output, err = createNewAllocationWithoutRetry(t, configPath, createParams(map[string]interface{}{"free_storage": markerFile}))
+		require.NotNil(t, err, "Failed to create new allocation", strings.Join(output, "\n"))
+		require.Greater(t, len(output), 1)
+		require.Equal(t, "Error creating free allocation: [txn] too less sharders to confirm it: min_confirmation is 50%, but got 0/2 sharders", output[0])
+	})
+
+	t.Run("Create free storage with tokens exceeding assigner's individual limit should fail", func(t *testing.T) {
+		t.Parallel()
+
+		recipient := escapedTestName(t)
+
+		// register recipient wallet
+		output, err = registerWalletForName(t, configPath, recipient)
+		require.Nil(t, err, "registering wallet failed", err, strings.Join(output, "\n"))
+
+		recipientWallet, err := getWalletForName(t, configPath, recipient)
+		require.Nil(t, err, "Error occurred when retrieving new owner wallet")
+
+		marker := climodel.FreeStorageMarker{
+			Recipient:  recipientWallet.ClientID,
+			FreeTokens: freeTokensIndividualLimit + 1,
+			Timestamp:  time.Now().Unix(),
+		}
+
+		forSignatureBytes, err := json.Marshal(&marker)
+		require.Nil(t, err, "Could not marshal marker")
+
+		data := hex.EncodeToString(forSignatureBytes)
+		marker.Signature = sign(t, data, assignerWallet)
+		marker.Assigner = assignerWallet.ClientID
+
+		forFileBytes, err := json.Marshal(marker)
+		require.Nil(t, err, "Could not marshal marker")
+
+		markerFile := "./config/" + recipient + "_MARKER.json"
+
+		err = os.WriteFile(markerFile, forFileBytes, 0600)
+		require.Nil(t, err, "Could not write file marker")
+
+		output, err = createNewAllocationWithoutRetry(t, configPath, createParams(map[string]interface{}{"free_storage": markerFile}))
+		require.NotNil(t, err, "Failed to create new allocation", strings.Join(output, "\n"))
+		require.Greater(t, len(output), 1)
+		require.Equal(t, "Error creating free allocation: [txn] too less sharders to confirm it: min_confirmation is 50%, but got 0/2 sharders", output[0])
+	})
+}
+
+func getZCNWallet(t *testing.T, file string) *climodel.WalletFile {
```

Comment:
The function name is ambiguous.  
```suggestion
func readWalletFile(t *testing.T, file string) *climodel.WalletFile {
```

Reply:
updated